crl/updater/updater.go

```diff
@@ -250,10 +259,12 @@ func (cu *crlUpdater) tickIssuer(ctx context.Context, atTime time.Time, issuerNa
 			case <-ctx.Done():
 				return
 			default:
+				ctx, cancel := context.WithTimeout(ctx, cu.updateTimeout)
 				out <- shardResult{
 					shardIdx: idx,
 					err:      cu.tickShardWithRetry(ctx, atTime, issuerNameID, idx, shardMap[idx]),
 				}
+				cancel()
```

Comment:
Because sending/receiving data from a channel blocks, we don't have to defer the cancel here, we know that as soon as the other side is ready we can cancel the context.

Reply:
And in fact we don't _want_ to `defer cancel()`, because this worker function might run for a long time (working on many shards one after another) and we'd rather not have all those clean-up cancels happen at the very end, they should be happening as we go.

Reply:
Ah so it's scoped to the surrounding function/goroutine rather than the switch statement level.
https://go.dev/ref/spec#Defer_statements